pkg/lib/oauth/handler/handler_authz.go

```diff
@@ -605,7 +633,22 @@ func (h *AuthorizationHandler) validateRequest(
 			return protocol.NewError("invalid_request", "only 'S256' PKCE transform is supported")
 		}
 	case NoneResponseType:
-		break
+		if err := requireScope(); err != nil {
+			return err
+		}
+	case AppInitiatedSSOToWebTokenResponseType:
+		if len(r.Prompt()) != 1 || r.Prompt()[0] != "none" {
+			return protocol.NewError("invalid_request", "only 'prompt=none' is supported when using app-initiated-sso-to-web")
+		}
+		if idTokenHint, ok := r.IDTokenHint(); !ok || idTokenHint == "" {
+			return protocol.NewError("invalid_request", "id_token_hint is required when using app-initiated-sso-to-web")
+		}
+		if r.AppInitiatedSSOToWebToken() == "" {
+			return protocol.NewError("invalid_request", "x_app_initiated_sso_to_web_token is required when using app-initiated-sso-to-web")
+		}
+		if r.ResponseMode() != "cookie" {
+			return protocol.NewError("invalid_request", "only 'response_mode=cookie' is supported when using app-initiated-sso-to-web")
+		}
 	default:
 		return protocol.NewError("unsupported_response_type", "only 'code' response type is supported")
```

Comment:
The message is now outdated. Maybe just output the raw response type we receive in the query (that is, the string, not the parsed response type).

```go
return protocol.NewError("unsupported_response_type", fmt.Sprintf("unsupported response_type: %v", rawResponseTypeString))
```